libsyclinterface/source/dpctl_sycl_extension_interface.cpp

```diff
@@ -62,3 +62,37 @@ bool DPCTLWorkGroupMemory_Available()
     return false;
 #endif
 }
+
+DPCTL_API
+__dpctl_give DPCTLSyclRawKernelArgRef DPCTLRawKernelArg_Create(void *bytes,
+                                                               size_t count)
+{
+    DPCTLSyclRawKernelArgRef rka = nullptr;
+    try {
+        auto rawData =
+            std::unique_ptr<unsigned char[]>(new unsigned char[count]);
+        std::memcpy(rawData.get(), bytes, count);
+        auto RawKernelArg = std::unique_ptr<RawKernelArgData>(
+            new RawKernelArgData{rawData.release(), count});
+        rka = wrap<RawKernelArgData>(RawKernelArg.release());
```

Comment:
Sorry. I've misled you in my previous comment about `unique_ptr`. To make it work it should be something like this:
```C++
        auto rawData =
            std::unique_ptr<unsigned char[]>(new unsigned char[count]);
        std::memcpy(rawData.get(), bytes, count);
        auto RawKernelArg = std::unique_ptr<RawKernelArgData>(
            new RawKernelArgData{rawData.get(), count});
        rka = wrap<RawKernelArgData>(RawKernelArg.get());
        RawKernelArg.release();
        rawData.release();
```

Otherwise `rawData` would leak if `new RawKernelArgData{rawData.release(), count}` throws.

Reply:
I changed the approach slightly to hold the data in a `std::vector` inside `RawKernelArgDataTy`.

Reply:
```std::vector``` is used indeed. The same as ```std::unique_ptr```. But it is in ```.cpp``` file. Not in header file which actually has ```.h``` extension and not ```.hpp``` and ```extern "C"``` declaration (https://github.com/IntelPython/dpctl/pull/2038/files#diff-bc26687692ae358c908d9587e3abdfab35454bc0fcab36f47cf59af185e6b5fdR40)